### DISCUSSION TITLE
[Script request] Unpackerr

### DISCUSSION THREAD

**Discussion by oOStroudyOo** [community-scripts/ProxmoxVE#969] [Request script] [answered]
### Application Name

Unpackerr

### Website

https://unpackerr.zip/ and https://github.com/Unpackerr/unpackerr

### Description

Unpackerr runs as a daemon on your download host or seedbox. It checks for completed downloads and extracts them so Lidarr, Radarr, Readarr and Sonarr may import them. If your problem is rar files getting stuck in your activity queue, then this is your solution.

Linux installer: https://unpackerr.zip/docs/install/linux

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by chroniclex** [comment]
I was looking for an LXC for this last weekend. even attempted to build my own. Hopefully this gets some support.

**Comment by alex8654** [comment]
I've been trying to make it work on a DIY with LXC, but my struggle is the permission mapping to make it work with a mounting point that qbittorrent downloads stuff to. Have you guys managed to get this working? Qbittorrent creates folders as root, but unpackerr creates folders as itself, and cannot deal with root permissions that qbittorrent uses.

**Reply by oOStroudyOo** [reply]
Hi @alex8654 

Are you trying to get this to work as an LXC script, or do you mean you're struggling to get Unpackerr to work at all?

**Reply by alex8654** [reply]
unpacker works inside LXC, my struggle is to understand permissions. qbittorrent LXC has access to the same mount point as unpackerr, but qbittorrent creates folders and files as root, so unpackerr does not have permission to create folder/files inside qbittorrent created folder to decompress files. I'm trying to find a way to allow unpacker to modify folders and files qbittorrent creates.

**Reply by oOStroudyOo** [reply]
I can't really remember.
You can change the user that Unpacker runs as with an override config for the service.

But I think I just set recersive permissions for the Unpackerr user to own all current and future files and folders.

I run this as a separate LXC to my qbittorrent.

**Reply by alex8654** [reply]
I ended up doing 777 permission to everything, it is just a media folder anyway, I also run everything separate to make easier to troubleshoot, one LXC per app

**Comment by MickLesk** [comment, marked as answer]
We have currently decided to reject the script. Reasons:
- Last release 2024

If a new release is created (because I can see commits there), feel free to push me again and I will review the request. @oOStroudyOo

**Reply by oOStroudyOo** [reply]
New version has now been released: https://github.com/Unpackerr/unpackerr

**Reply by MickLesk** [reply]
okay, i take a look

**Comment by alex8654** [comment]
I'm, not sure any meaningful fix has been needed since then, it is a very stable and mature project. Why would a newer release be mandatory?

**Reply by MickLesk** [reply]
very old deps inside the tarball / releases

**Reply by alex8654** [reply]
thanks for letting me know, I'll talk to the guys on the project to see if we can get the ball rolling on a new version with updated dependencies.
